src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -292,20 +337,79 @@ void java_bytecode_convert_classt::convert(const classt &c)
     throw 0;
   }
 
-  // now do fields
+  // Now do fields
+  for(auto overlay_class : overlay_classes)
+  {
+    for(const auto &field : overlay_class.get().fields)
+    {
+      debug()
+        << "Adding symbol from overlay class:  field '" << field.name << "'"
+        << eom;
+      convert(*class_symbol, field);
+    }
+  }
   for(const auto &field : c.fields)
```

Comment:
I would add the "real" fields first, then if an overlay field is going to hide an existing field can drop a warning, unless there is a good reason not to.

Reply:
Fields like `@class_identifier` will always be duplicated and can't be marked as overlays. I am happy that the final overlay gets to specify the final type/value of the fields, though we should probably warn if the types don't match.